### ISSUE TITLE
Features of autograd.record

### ISSUE THREAD

**Issue by two-names** [apache/incubator-mxnet#14564]
Hello!

What could be the problem?
I have saved gluon model after 250 epochs like (model.save_parameters(...)).
After that I try to get output from load_params model:

First way is:
```
with autograd.record():
    outputs = [model(X) for X in data]
```
Second way is just:
`outputs = [model(X) for X in data]  ` 

First way gives better predictions.

What could be the reason?

**Comment by ptrendx** [comment]
If you have e.g. batchnorm in your model, the first way will use training mode of it (using statistics from your batch), whereas the second way will call inference mode (which uses statistics gathered during training).

**Comment by two-names** [comment]
@ptrendx 
But default use_global_stats is False in all BN in my model. It's meaning that global moving statistics is not agreagating.

**Comment by ptrendx** [comment]
No. `use_global_stats=True` would mean that you don't calculate statistics during training, `False` calculates statistic from a current batch. In inference mode you never calculate statistics, and instead always use the global mean and variance.

**Comment by two-names** [comment]
@ptrendx thanks!

**Comment by haojin2** [comment]
@two-names Please feel free to re-open the issue if you have further questions, closing for now.
